### DISCUSSION TITLE
Execution path issue with systemd

### DISCUSSION THREAD

**Discussion by gnh1201** [misskey-dev/misskey#13987] [General]
## 💡 Summary

I recently found out about this problem on my Mastodon server. https://github.com/mastodon/mastodon/issues/20671#issuecomment-1358980675

After I found this problem, I tried installing Misskey and found out that there was the same problem.

Mastodon was able to force a change in the execution path, but Misskey was unable to.

## 🥰 Expected Behavior

After you forcefully change the execution path, the systemd daemon must run normally. For example:

```diff
[Unit]
Description=Misskey daemon

[Service]
Type=simple
User=misskey
-ExecStart=/usr/bin/npm start
+ExecStart=/home/misskey/.nvm/versions/node/v16.19.0/bin/npm start
WorkingDirectory=/home/misskey/misskey
Environment="NODE_ENV=production"
TimeoutSec=60
StandardOutput=syslog
StandardError=syslog
SyslogIdentifier=misskey
Restart=always

[Install]
WantedBy=multi-user.target
```

## 🤬 Actual Behavior

The daemon is not functioning properly with reference to the wrong node and npm during subprocess creation. This causes the script to rerun indefinitely.

But there's no problem doing this manually in the command line: `NODE_ENV=production npm start`

## 📝 Steps to Reproduce

1. Create new `/etc/systemd/system/misskey.service` file and modify it. (Change the exectation path)
2. `systemctl daemon-reload` and `systemctl start misskey`
3. `systemctl start misskey`

## 📌 Environment

Misskey version: 12.119.2
Your OS: Debian 11
Your browser: Microsoft Edge 108.0.1462.54

**Comment by acid-chicken** [comment]
The installation instructions provided in the Misskey Hub assume a direct Node.js installation and cannot be built as documented in an environment using nvm.
Also, you are using nvm incorrectly, the following gist may be helpful on how to use nvm with systemd: https://gist.github.com/joepie91/73ce30dd258296bd24af23e9c5f761aa

**Comment by gnh1201** [comment]
Thank you for your advice. However, I have already used this method, but the process fork is not occurring in the correct path. I think I should use it manually in the command line for the time being.

> The installation instructions provided in the Misskey Hub assume a direct Node.js installation and cannot be built as documented in an environment using nvm. Also, you are using nvm incorrectly, the following gist may be helpful on how to use nvm with systemd: https://gist.github.com/joepie91/73ce30dd258296bd24af23e9c5f761aa
